app/assets/javascripts/govuk_publishing_components/components/feedback.js

```diff
@@ -0,0 +1,153 @@
+window.GOVUK = window.GOVUK || {};
+window.GOVUK.Modules = window.GOVUK.Modules || {};
+
+(function (Modules) {
+  "use strict";
+  window.GOVUK = window.GOVUK || {};
+
+  Modules.Feedback = function () {
+
+    this.start = function ($element) {
+      this.$prompt = $element.find('.js-prompt');
+      this.$fields = $element.find('.pub-c-feedback__form-field');
+      this.$forms = $element.find('.js-feedback-form');
+      this.$toggleForms = $element.find('.js-toggle-form');
+      this.$closeForms = $element.find('.js-close-form');
+      this.$activeForm = false;
+      this.$pageIsUsefulButton = $element.find('.js-page-is-useful');
+      this.$pageIsNotUsefulButton = $element.find('.js-page-is-not-useful');
+      this.$somethingIsWrongButton = $element.find('.js-something-is-wrong');
+      this.$promptQuestions = $element.find('.js-prompt-questions');
+      this.$promptSuccessMessage = $element.find('.js-prompt-success');
+
+      var that = this;
+
+      setInitialAriaAttributes();
+
+      this.$toggleForms.on('click', function(e) {
+        e.preventDefault();
+        toggleForm($(e.target).attr('aria-controls'));
+        trackEvent(getTrackEventParams($(this)));
+        updateAriaAttributes($(this));
+      });
+
+      this.$closeForms.on('click', function(e) {
+        e.preventDefault();
+        toggleForm($(e.target).attr('aria-controls'));
+        setInitialAriaAttributes();
+      });
+
+      this.$pageIsUsefulButton.on('click', function(e) {
+        e.preventDefault();
+        trackEvent(getTrackEventParams(that.$pageIsUsefulButton));
+        showFormSuccess();
+        revealInitialPrompt();
+      });
+
+      $element.find('form').on('submit', function(e) {
+        e.preventDefault();
+        var $form = $(this);
+        console.log($form.serialize())
+        $.ajax({
```

Comment:
What happens if this ajax call never resolves? Does it timeout? Or does the user sit waiting forever?

Reply:
That's an excellent question. Ideally I guess it times out... how best to implement that?